sdk/tests/NodeAddressTest.cc

```diff
@@ -0,0 +1,151 @@
+}
+
+// Test creation of NodeAddress instance using a protobuf object.
+TEST_F(NodeAddressTest, ConstructFromProtobuf)
+{
+  // Given
+  const int testPortPlain = getTestPortPlain();
+  const int64_t testNodeId = getTestNodeId();
+  const std::string& testRSAPublicKey = getTestRSAPublicKey();
+  const std::string& testIpAddressV4 = getTestIpAddress();
+  const std::string& testDescription = getTestDescription();
+  proto::NodeAddress testProtoNodeAddress = proto::NodeAddress();
+  testProtoNodeAddress.set_nodeid(testNodeId);
+  testProtoNodeAddress.set_description(testDescription);
+  testProtoNodeAddress.set_rsa_pubkey(testRSAPublicKey);
+  proto::ServiceEndpoint* serviceEndpoint = testProtoNodeAddress.add_serviceendpoint();
+  serviceEndpoint->set_ipaddressv4(testIpAddressV4);
+  serviceEndpoint->set_port(testPortPlain);
+
+  // When
+  const NodeAddress nodeAddress = NodeAddress::fromProtobuf(testProtoNodeAddress);
+
+  // Then
+  EXPECT_EQ(nodeAddress.getDefaultIpAddress().toString(), testIpAddressV4);
+  EXPECT_EQ(nodeAddress.getDefaultPort(), getTestPortTLS());
+  EXPECT_EQ(nodeAddress.getNodeId(), testNodeId);
+  EXPECT_EQ(nodeAddress.getPublicKey(), testRSAPublicKey);
+  EXPECT_EQ(nodeAddress.getNodeAccountId().getShardNum(), 0ULL);
+  EXPECT_EQ(nodeAddress.getNodeAccountId().getRealmNum(), 0ULL);
+  EXPECT_FALSE(nodeAddress.getNodeAccountId().getAccountNum().has_value());
+  EXPECT_EQ(nodeAddress.getNodeAccountId().getAlias(), nullptr);
+  EXPECT_FALSE(nodeAddress.getNodeAccountId().getEvmAddress().has_value());
+  EXPECT_TRUE(nodeAddress.getNodeCertHash().empty());
+  EXPECT_FALSE(nodeAddress.getDescription().empty());
+  EXPECT_EQ(nodeAddress.getDescription(), testDescription);
+  EXPECT_FALSE(nodeAddress.getEndpoints().empty());
+}
+
+// Test creation of NodeAddress instance using a given string node address.
+TEST_F(NodeAddressTest, ConstructFromString)
+{
+  // Given
+  const int testPort = getTestPortTLS();
+  const std::string& testIpAddressV4 = getTestIpAddress();
+  const std::string testNodeAddress = getTestNodeAddress();
+
+  // When
+  const NodeAddress nodeAddress = NodeAddress::fromString(testNodeAddress);
+
+  // Then
+  EXPECT_EQ(nodeAddress.getDefaultIpAddress().toString(), testIpAddressV4);
+  EXPECT_EQ(nodeAddress.getDefaultPort(), testPort);
+  EXPECT_EQ(nodeAddress.getNodeId(), -1);
+  EXPECT_FALSE(nodeAddress.getNodeAccountId().getEvmAddress().has_value());
+  EXPECT_EQ(nodeAddress.getNodeAccountId().getShardNum(), 0ULL);
+  EXPECT_EQ(nodeAddress.getNodeAccountId().getRealmNum(), 0ULL);
+  EXPECT_FALSE(nodeAddress.getNodeAccountId().getAccountNum().has_value());
+  EXPECT_EQ(nodeAddress.getNodeAccountId().getAlias(), nullptr);
+  EXPECT_FALSE(nodeAddress.getNodeAccountId().getEvmAddress().has_value());
+  EXPECT_TRUE(nodeAddress.getNodeCertHash().empty());
+  EXPECT_TRUE(nodeAddress.getDescription().empty());
+  EXPECT_FALSE(nodeAddress.getEndpoints().empty());
+}
```

Comment:
You should add small tests for the getter and setter functions for `NodeAddress` as well.